wolfcrypt/src/aes.c

```diff
@@ -9311,8 +9315,91 @@ int wc_AesGcmInit(Aes* aes, const byte* key, word32 len, const byte* iv,
     if (ret == 0) {
         /* Setup with IV if needed. */
         if (iv != NULL) {
-            /* Cache the IV in AES GCM object. */
-            XMEMCPY((byte*)aes->reg, iv, ivSz);
+            /* Initialize and generate GHASH from provided IV. */
+            ret = AesGcmInit_C(aes, iv, ivSz);
+            if (ret == 0) {
+                /* Cache the GHASH in the AES GCM object. */
+                XMEMCPY((byte*)aes->reg, AES_COUNTER(aes), sizeof(aes->reg));
+                aes->nonceSz = (ivSz > sizeof(aes->reg) ?
+                                sizeof(aes->reg) : ivSz);
+            }
+        }
+        else if (aes->nonceSz != 0) {
+            /* Copy out the cached copy. */
+            iv = (byte*)aes->reg;
+            ivSz = aes->nonceSz;
+            needInit = 1;
+        }
+
+        if ((ret == 0) && (iv != NULL)) {
+            /* Initialize with the IV. */
+        #ifdef WOLFSSL_AESNI
+            if (haveAESNI
+            #ifdef HAVE_INTEL_AVX2
+                || IS_INTEL_AVX2(intel_flags)
+            #endif
+            #ifdef HAVE_INTEL_AVX1
+                || IS_INTEL_AVX1(intel_flags)
+            #endif
+                ) {
+                ret = AesGcmInit_aesni(aes, iv, ivSz);
+            }
+            else
+        #endif
+            if (needInit) {
+                ret = AesGcmInit_C(aes, iv, ivSz);
+            }
+
+            aes->nonceSet = 1;
+        }
+    }
+
+    return ret;
+
+#endif // TWW
+
+#if 0 // TWW - Use GHASH - EXPERIMENTAL (not working)
```

Comment:
When I tried to directly use the `GHASH` to condense the IV, it did not work.  I included the WIP code I had here in case it was close, and just needed some adjustments...